src/zenml/cli/hub.py

```diff
@@ -0,0 +1,1203 @@
+)
+@click.option(
+    "--repository_branch",
+    "-b",
+    type=str,
+    help="Branch to checkout from the repository.",
+)
+@click.option(
+    "--repository_commit",
+    "-c",
+    type=str,
+    help="Commit to checkout from the repository. Overrides --branch.",
+)
+@click.option(
+    "tags",
+    "--tag",
+    "-t",
+    type=str,
+    multiple=True,
+)
+@click.option(
+    "--interactive",
+    "-i",
+    is_flag=True,
+    help="Run the command in interactive mode.",
+)
+def submit_plugin(
+    plugin_name: Optional[str],
+    version: Optional[str],
+    release_notes: Optional[str],
+    description: Optional[str],
+    repository_url: Optional[str],
+    repository_subdir: Optional[str],
+    repository_branch: Optional[str],
+    repository_commit: Optional[str],
+    tags: List[str],
+    interactive: bool,
+) -> None:
+    """Submit a plugin to the ZenML Hub.
+
+    Args:
+        plugin_name: Name of the plugin to submit. Needs to be set unless
+            interactive mode is enabled.
+        version: Version of the plugin to submit. Can only be set if the plugin
+            already exists. If not provided, the version will be
+            auto-incremented.
+        release_notes: Release notes for the plugin version.
+        description: Description of the plugin.
+        repository_url: URL to the public Git repository containing the plugin
+            source code. Needs to be set unless interactive mode is enabled.
+        repository_subdir: Subdirectory of the repository containing the plugin
+            source code.
+        repository_branch: Branch to checkout from the repository.
+        repository_commit: Commit to checkout from the repository. Overrides
+            `repository_branch`.
+        tags: Tags to add to the plugin.
+        interactive: Whether to run the command in interactive mode, asking for
+            missing or invalid parameters.
+    """
+    with event_handler(
+        event=AnalyticsEvent.ZENML_HUB_PLUGIN_SUBMIT,
+        metadata={
+            "plugin_name": plugin_name,
+            "plugin_version": version,
+            "repository_url": repository_url,
+        },
+    ) as analytics_handler:
+        client = HubClient()
+        analytics_handler.metadata["hub_url"] = client.url
+
+        # Validate that the user is logged in
+        if not client.auth_token:
+            error(
+                "You must be logged in to contribute a plugin to the Hub. "
+                "Please run `zenml hub login` to login."
+            )
+        me = client.get_me()
+        if not me:
+            error("Could not retrieve user information from the ZenML Hub.")
+        if not me.username:
+            error(
+                "Your ZenML Hub account does not have a username yet. Please "
+                "set a username in your account settings and try again."
+            )
+        username = me.username
+        analytics_handler.metadata["plugin_author"] = username
```

Comment:
https://docs.zenml.io/misc/usage-analytics

In order to help us better understand how the community uses ZenML, the pip package reports **anonymized** usage statistics.

Reply:
oh yes, fair enough, I'll remove the author tracking for plugin create. 

The other CLI commands also have the plugin author in the analytics metadata (since the plugin name alone is not unique), I guess that is fine since it is information of the plugin that is interacted with and not information of the active user at that point?

Reply:
Yep, I guess the usage of a plugin (e.g. downloads/installs) is even something that's publicly displayed right?